core/src/main/java/org/lflang/generator/c/CCmakeGenerator.java

```diff
@@ -519,4 +540,134 @@ private static String setUpMainTargetRp2040(
     code.newLine();
     return code.toString();
   }
-}
+
+
+  private static String setUpMainTargetStm32(
+          boolean hasMain, String executableName, Stream<String> cSources, String[] boardProperties) {
+    var code = new CodeBuilder();
+    code.pr("\n# ########################################################");
+    code.pr("\n# ############## [ Start STM32 main target] ##############");
+    code.pr("\n# ########################################################");
+    code.newLine();
+    code.newLine();
+
+    code.pr("add_subdirectory(core)");
+    code.newLine();
+    code.pr("set(LF_MAIN_TARGET " + executableName + ")");
+
+    // Declaration preamble
+    code.pr("set(PROJECT_DIR ${CMAKE_CURRENT_SOURCE_DIR})");
+    code.pr("set(STM_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../STM_sdk)");
+    code.newLine();
+
+    // Configure board settings
+    if (boardProperties.length < 1 || boardProperties[0].equals("")) {
+      // By default, we set it to STM32f446RE
+      code.pr("set(MCU_FAMILY STM32F4xx)");
+      code.pr("set(MCU_MODEL STM32F446xx)");
+    } else {
+      code.pr("set(MCU_FAMILY STM32" + boardProperties[0].substring(0, 2) + "xx)");
+      code.pr("set(MCU_MODEL STM32" + boardProperties[0].substring(0, 4) + "xx)");
+    }
+
+
+    code.pr("set(CPU_PARAMETERS -mcpu=cortex-m4 -mthumb -mfpu=fpv4-sp-d16 -mfloat-abi=softfp)");
```

Comment:
It seems that these parameters should depend on the chosen platform.

Reply:
These might actually be commen for all STM32s